packages/flex-plugin-scripts/src/clients/build.ts

```diff
@@ -0,0 +1,75 @@
+import { AuthConfig } from './auth';
+import BaseClient from './baseClient';
+import { Build, BuildStatus } from './serverless-types';
+import ServiceClient from './services';
+
+export interface BuildData {
+  FunctionVersions: string[];
+  AssetVersions: string[];
+  Dependencies: object;
+}
+
+export default class BuildClient extends BaseClient {
+  private static timeoutMsec: number = 60000;
+  private static pollingIntervalMsec: number = 500;
+
+  constructor(auth: AuthConfig, serviceSid: string) {
+    super(auth,  `${ServiceClient.getBaseUrl()}/Services/${serviceSid}`);
+  }
+
+  /**
+   * Creates a new {@link Build} and then polls the endpoint once a second until the build is
+   * complete.
+   *
+   * @param data  the build data
+   */
+  public create = (data: BuildData): Promise<Build> => {
+    return new Promise(async (resolve, reject) => {
+      const newBuild = await this._create(data);
+      const sid = newBuild.sid;
```

Comment:
`const { sid } = newBuild;`

Reply:
For deconstructing a single property, don't see the diff/advantage of one over another.

```js
const sid = newBuild.sid;
const { sid } = newBuild;
```

Reply:
just cleaner imo